File: track_loader/track.py
class Track:
    pilotname = ""
    location = ""
    altitude = 0
    duration = 0
    distance = 0
    trackid = 0
    lasttime = ""
    activity = ""
    isLive = False

    def get_metadata(self):
        return {
            'pilotname': self.pilotname,
            'location': self.location,
            'duration': self.duration,
            'distance': self.distance,
            'trackid': self.trackid,
            'lasttime': self.lasttime,
            'activity': self.activity,
        }

    def filename(self):
        return self.pilotname + "_" + self.lasttime.strftime("%Y%m%d%H%M%S")

    def __str__(self):
        return "pilotname: " + self.pilotname + " location: " + self.location + " id: " + str(self.trackid) + " lasttime: " + str(self.lasttime)

File: track_loader/track.js

class Track {
    #pilotname
    #liveTrackId
    #isLive

    constructor(pilotname, liveTrackId, isLive) {
        this.#pilotname = pilotname;
        this.#liveTrackId = liveTrackId;
        this.#isLive = isLive;
    }

    get pilotname() {
        return this.#pilotname;
    }
    get liveTrackId() {
        return this.#liveTrackId;
    }
    get isLive() {
        return this.#isLive;
    }
}

module.exports = { Track };